### PR TITLE
[AV-68571] Vector Search - The Capella Files

### PR DESCRIPTION
**Pull request by sarahlwelton**
This PR contains all of the documentation changes required for Vector Search on Capella. 

This PR will remain in draft until more review comments have been received. 

View the contents of the PR in context here: https://preview.docs-test.couchbase.com/sarah-AV-68571/cloud/vector-search/vector-search.html 

Username and password is the same as for all docs preview sites: https://couchbasecloud.atlassian.net/wiki/spaces/DOCS/pages/1971224949/Docs+Team+demo+site+passwords

### CODE REVIEW

File: modules/n1ql/pages/n1ql-language-reference/searchfun.adoc [hunk 1]
```diff
@@ -572,4 +614,6 @@ LIMIT 3;
 
 == Related Links
 
-xref:server:fts:full-text-intro.adoc[Full Text Search]
+xref:search:search.adoc[]
+xref:vector-search:vector-search.adoc
```
Comment (dgnicholson) [inline modules/n1ql/pages/n1ql-language-reference/searchfun.adoc]: Broken links / missing bullets: 

```suggestion
* xref:search:search.adoc[]
* xref:vector-search:vector-search.adoc[]
```

File: modules/search/partials/vector-search-field-descriptions.adoc [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+NOTE: It's recommended to choose the same similarity metric for your Search index as the one used in your embedding model.
+
+* *dot_product*: Calculated by adding the result of multiplying a vector's components, or the product of the magnitudes of the vectors and the cosine of the angle between them.
+The dot product of 2 vectors is affected by the length and direction of each of the vectors, rather than just taking a straight-line distance. 
++
+Dot product similarity is commonly used by Large Language Models (LLMs).
+Use *dot_product* to get the best results with an embedding model that uses dot product similarity. 
+
+* *l2_norm*: Also known as euclidean distance.
```
Comment (dgnicholson) [inline modules/search/partials/vector-search-field-descriptions.adoc]: Super Tiny Nit: I think Euclidean should be capitalized, as it's derived from the mathematician's name.

File: modules/vector-search/pages/vector-search.adoc [hunk 3]
```diff
@@ -0,0 +1,49 @@
+= Use Vector Search for AI Applications
```
Comment (dgnicholson) [inline modules/vector-search/pages/vector-search.adoc:1]: Simple, but great title. 🚀

File: modules/vector-search/pages/vector-search.adoc [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+:page-topic-type: concept
+:description: Use Couchbase Capella's Vector Search features to add fast and accurate semantic search to your applications.
+
+[abstract]
+{description}
+
+== About Vector Search
+
+Vector Search builds on Couchbase Capella's xref:search.adoc[Search Service] to provide vector index support.
```
Comment (dgnicholson) [inline modules/vector-search/pages/vector-search.adoc]: Broken link
```suggestion
Vector Search builds on Couchbase Capella's xref:search:search.adoc[Search Service] to provide vector index support.
```

File: modules/vector-search/pages/vector-search.adoc [hunk 5]
```diff
@@ -0,0 +1,49 @@
+= Use Vector Search for AI Applications
+:page-topic-type: concept
+:description: Use Couchbase Capella's Vector Search features to add fast and accurate semantic search to your applications.
+
+[abstract]
+{description}
+
+== About Vector Search
```
Comment (dgnicholson) [inline modules/vector-search/pages/vector-search.adoc:10]: This section is great. It's really hard to summarize these kinds of features w/o getting into the weeds, but you did it very well IMO.

File: modules/vector-search/partials/nav.adoc [hunk 6]
```diff
@@ -0,0 +1,5 @@
+* xref:cloud:vector-search:vector-search.adoc[]
+** xref:cloud:vector-search:create-vector-search-index-ui.adoc[]
+** xref:cloud:vector-search:create-vector-search-index-rest-api.adoc[]
+** xref:cloud:vector-search:run-vector-search-ui.adoc[]
+** xref:cloud:vector-search:run-vector-search-rest-api.adoc[]
```
Comment (dgnicholson) [inline modules/vector-search/partials/nav.adoc:5]: Out of curiosity, what was your reasoning on dividing the API instructions and the UI instructions into different pages and not do something like tabs?

File: modules/vector-search/pages/create-vector-search-index-ui.adoc [hunk 7]
```diff
@@ -0,0 +1,74 @@
+= Create a Vector Search Index in Quick Mode
+:page-topic-type: guide
+:description: Use Quick Mode to create a Vector Search index in Couchbase Capella.
+
+[abstract]
+{description}
+
+Use xref:create-search-index-ui.adoc[Advanced Mode] to add additional settings and more advanced options for your Vector Search index and xref:customize-index.adoc#type-mappings[type mappings].
```
Comment (dgnicholson) [inline modules/vector-search/pages/create-vector-search-index-ui.adoc]: Broken links (I think it's supposed to be in the `:search:` directory?)

File: modules/vector-search/pages/vector-search.adoc [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+Using vector embeddings to represent data, your end users can search based on descriptions, rather than using specific keywords.
+
+* *Generative AI:* Create new text, images, and more based on a prompt given to the LLM.  
+
+Vector Search supports integrations with frameworks like https://python.langchain.com/docs/get_started/introduction[LangChain^] to support AI application development. 
+For more information about all frameworks and integrations supported by Vector Search and Capella, see xref:third-party:integrations.adoc[].
+
+== Using Vector Search Indexes
+
+Use a Capella database to store the data you want to use to provide context to a LLM.
```
Comment (lamagnifica) [inline modules/vector-search/pages/vector-search.adoc]: I'm finding the command structure of these first three sentneces odd. Maybe frame as, Using VSIs is a three-step process: 
1.
2.
3...

File: modules/vector-search/pages/vector-search.adoc [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+For more information about all frameworks and integrations supported by Vector Search and Capella, see xref:third-party:integrations.adoc[].
+
+== Using Vector Search Indexes
+
+Use a Capella database to store the data you want to use to provide context to a LLM. 
+Generate vector embeddings using your preferred embedding model, then store the embeddings in an array inside your documents. 
+
+Create a Vector Search index to use the vector embeddings from your data and find matches from a search query's generated vectors.
+
+If you do not want to use an integration like LangChain, you can also use the API for an existing LLM and one of their embedding models to generate these embeddings for your data.
```
Comment (lamagnifica) [inline modules/vector-search/pages/vector-search.adoc]: Maybe reframe as a positive, "In addition to supporting integrations with frameworks like LangChain, you can also...

File: modules/vector-search/pages/create-vector-search-index-ui.adoc (hunk 7 above)
Comment (lamagnifica) [inline modules/vector-search/pages/create-vector-search-index-ui.adoc]: incorrect links? both fail in the preview

File: modules/vector-search/pages/create-vector-search-index-ui.adoc [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+
+== Prerequisites 
+
+* You have the Search Service enabled on a node in your database.
+For more information about how to change Services on your database, see xref:cloud:clusters:modify-database.adoc[].
+
+* You have a bucket with scopes and collections in your database. 
+For more information, see xref:cloud:clusters:data-service/manage-buckets.adoc[].
+
+* You have documents in a keyspace inside that bucket that contain vector embeddings.
```
Comment (dgnicholson) [inline modules/vector-search/pages/create-vector-search-index-ui.adoc]: Nit: I tripped over this line a little bit. I'm wondering if we change "that bucket" to "your bucket" would help.

```suggestion
* You have documents in a keyspace inside your bucket that contain vector embeddings.
```

File: modules/vector-search/pages/create-vector-search-index-ui.adoc (hunk 7 above)
Comment (lamagnifica) [reply modules/vector-search/pages/create-vector-search-index-ui.adoc]: which makes me wonder, does this quick mode topic also cover these advanced procedures? if not, maybe
After you create a VSI using quick made, you can configure more settings and add more advanced options for your Vector Search index and type mappings

File: modules/vector-search/pages/create-vector-search-index-ui.adoc [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+. Expand the collection that contains these documents. 
+. In your document schema, select a field that contains your vector embeddings. 
++
+Vectors should be an array of floating point numbers.
+. Configure the options for your new child field as follows:
+.. In the *Type* list, select *vector*. 
+.. In the *Dimension* field, check that the value matches the total number of elements in your vector field's array. 
++
+The Search Service supports arrays up to 2048 elements.
+Capella should automatically fill in the dimension value for your selected array field.
```
Comment (dgnicholson) [inline modules/vector-search/pages/create-vector-search-index-ui.adoc]: Nit: perhaps make this a little less passive in tone—makes it sounds like it could break (I mean sure it can 😆 , but we don't want to convey that).

```suggestion
Capella automatically fills in the dimension value for your selected array field.
```

File: modules/vector-search/pages/create-vector-search-index-ui.adoc [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+.. In the *Similarity metric* list, select the method to use to calculate the similarity between search term and Search index vectors.
++ 
+For more information, see xref:search:quick-index-field-options.adoc#quick-field[Quick Field Type Mapping Options].
+.. In the *Optimized for* list, select whether the Search Service should optimize Search queries for accuracy (*recall*) or speed (*latency*).
++
+For more information, see xref:xref:search:quick-index-field-options.adoc#quick-field[Quick Field Type Mapping Options].
```
Comment (dgnicholson) [inline modules/vector-search/pages/create-vector-search-index-ui.adoc]:
```suggestion
For more information, see xref:search:quick-index-field-options.adoc#quick-field[Quick Field Type Mapping Options].
```

File: modules/vector-search/pages/create-vector-search-index-ui.adoc [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+Your index name must start with an alphabetic character (a-z or A-Z). It can only contain alphanumeric characters (a-z, A-Z, or 0-9), hyphens (-), or underscores (_).
+
+For version 7.6 and later, your index name must be unique inside your selected bucket and scope. You cannot have 2 indexes with the same name inside the same bucket and scope.
+====
+. Under *Type Mappings*, in the *Bucket* list, select the bucket that contains the documents you want to include in your index. 
+. In the *Scope* list, select the scope that contains these documents. 
+. Expand the collection that contains these documents. 
+. In your document schema, select a field that contains your vector embeddings. 
++
+Vectors should be an array of floating point numbers.
```
Comment (lamagnifica) [inline modules/vector-search/pages/create-vector-search-index-ui.adoc]: maybe mention this in the prerequisites section?

Comment (lamagnifica) [reply modules/vector-search/pages/create-vector-search-index-ui.adoc]: also, is should the right word? are there workarounds if they are not floating point numbers, or in an array?

File: modules/vector-search/pages/create-vector-search-index-ui.adoc [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+. Configure the options for your new child field as follows:
```
Comment (lamagnifica) [inline modules/vector-search/pages/create-vector-search-index-ui.adoc:59]: my what now? I missed where a new field got created. Maybe this is evident when actually using the UI, but perhaps state that "select a field that contains your vector embeddings" in the previous step has the result of adding a field to the schema ?

File: modules/vector-search/pages/create-vector-search-index-ui.adoc [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+For more information, see xref:xref:search:quick-index-field-options.adoc#quick-field[Quick Field Type Mapping Options].
+.. Select *Index*.
+. Click btn:[Submit].
+. (Optional) Add additional collections or field type mappings to your index. 
+. Click btn:[Create Index].
+
+== Next Steps 
+
+This basic Vector Search index includes the vector embeddings from the field you specified in your type mapping.
+If you chose to add additional child fields, the Search Service can also return data from those fields when you run a Vector Search query.
```
Comment (lamagnifica) [inline modules/vector-search/pages/create-vector-search-index-ui.adoc]: the references to child fields are quite confusing to me. Does optional step 11 add these child fields?

File: modules/vector-search/pages/create-vector-search-index-ui.adoc [hunk 16]
```diff
@@ -0,0 +1,74 @@
...
++
+For more information, see xref:xref:search:quick-index-field-options.adoc#quick-field[Quick Field Type Mapping Options].
+.. Select *Index*.
+. Click btn:[Submit].
+. (Optional) Add additional collections or field type mappings to your index. 
+. Click btn:[Create Index].
+
+== Next Steps 
+
+This basic Vector Search index includes the vector embeddings from the field you specified in your type mapping.
```
Comment (lamagnifica) [inline modules/vector-search/pages/create-vector-search-index-ui.adoc]: is the type mapping what I did in step 9? 
is this from the specified field or from the "child field" that got created?

File: modules/vector-search/pages/create-vector-search-index-rest-api.adoc [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+You must create a Search index before you can xref:run-vector-search-rest-api.adoc[run a search] with the Search Service.
+// Add link
+
+== Prerequisites 
+
+* You have the Search Service enabled on a node in your database. 
+
+* You have a bucket with scopes and collections in your database. 
+
+* You have documents in a keyspace inside that bucket that contain vector embeddings.
```
Comment (dgnicholson) [inline modules/vector-search/pages/create-vector-search-index-rest-api.adoc]: Nit: I wonder if the language of these prereq items should be reconciled more with the same ones in Quick Mode procedure.

File: modules/vector-search/pages/create-vector-search-index-ui.adoc [hunk 18]
```diff
@@ -0,0 +1,74 @@
...
+If you chose to add additional child fields, the Search Service can also return data from those fields when you run a Vector Search query. 
+
+You can customize your Vector Search index like any other Search index to add additional data and improve search results. 
+For more information about how to customize an index, see xref:search:customize-index.adoc[].
+
+CAUTION: Some Search index features are only available in Advanced Mode. If you edit your Search index in Advanced Mode, you cannot return to Quick Mode and keep any Advanced Mode settings.
```
Comment (lamagnifica) [inline modules/vector-search/pages/create-vector-search-index-ui.adoc]: instead of "return to Quick Mode" I'd rephrase more like, 
After you use Advanced Mode to configure your Search Index, you cannot use Quick Mode for any further edits. Using Quick Mode resets all Advanced Mode settings to their default state.
...If that is, in fact, what happens...

File: modules/vector-search/pages/create-vector-search-index-rest-api.adoc [hunk 19]
```diff
@@ -0,0 +1,59 @@
...
+
+* You have the hostname for your database.
++
+For more information about how to find the hostname for your Capella database, see xref:clouds:get-capella-host-name.adoc[].
+
+== Procedure 
+
+To create a Search index with the REST API and Couchbase Capella: 
+
+. In your command-line tool, enter a `curl` command with the `XPUT` verb.
```
Comment (lamagnifica) [inline modules/vector-search/pages/create-vector-search-index-rest-api.adoc:35]: verb --> method?

File: modules/vector-search/pages/create-vector-search-index-rest-api.adoc [hunk 20]
```diff
@@ -0,0 +1,59 @@
...
++
+For more information about how to find the hostname for your Capella database, see xref:clouds:get-capella-host-name.adoc[].
+
+== Procedure 
+
+To create a Search index with the REST API and Couchbase Capella: 
+
+. In your command-line tool, enter a `curl` command with the `XPUT` verb. 
+. Set your header content to include `Content-Type: application/json`.
+. Enter your username, password, and the Search Service endpoint on port `8094` with the name of the index you want to create:
```
Comment (lamagnifica) [inline modules/vector-search/pages/create-vector-search-index-rest-api.adoc]: ...with the ^fully-qualified^ name of the index...

File: modules/vector-search/pages/create-vector-search-index-rest-api.adoc [hunk 21]
```diff
@@ -0,0 +1,59 @@
...
+----
+include::example$create-vector-search-index-header.sh[]
+----
+. Enter the JSON payload for the settings you want in your index. 
++
+Do not include the xref:search:search-index-params.adoc#uuid[uuid] or xref:search:search-index-params.adoc#sourceuuid[sourceUUID] parameters. 
++
+TIP: If you remove the xref:search:search-index-params.adoc#uuid[uuid] and xref:search:search-index-params.adoc#sourceuuid[sourceUUID] parameters, you can copy the Search index definition JSON payload from the Capella UI to use in a REST API call.
+For more information about how to create a Vector Search index with the UI, see xref:create-vector-search-index-ui.adoc[]. 
++
```
Comment (lamagnifica) [inline modules/vector-search/pages/create-vector-search-index-rest-api.adoc]: Maybe instead of indenting under the final step, make an  .Example or ==Example header to call more attention to this?

File: modules/vector-search/pages/run-vector-search-ui.adoc [hunk 22]
```diff
@@ -0,0 +1,43 @@
...
+----
++
+By using the special `match_none` query in the `query` field, the Search query only returns the `k` number of similar vectors.
+
+. Press kbd:[Enter].
+. (Optional) To view a document and its source collection, click a document name in the search results list. 
+
+== Next Steps 
+
+If you do not get the search results you were expecting, you can change the xref:search-request-params.adoc[JSON payload for your Search query].
```
Comment (dgnicholson) [inline modules/vector-search/pages/run-vector-search-ui.adoc]:
```suggestion
If you do not get the search results you were expecting, you can change the xref:search-request-params.adoc[JSON payload for your Search query].
```
```suggestion
If you do not get the search results you were expecting, you can change the xref:search:search-request-params.adoc[JSON payload for your Search query].
```

File: modules/vector-search/pages/create-vector-search-index-rest-api.adoc [hunk 23]
```diff
@@ -0,0 +1,59 @@
...
+[source,console]
+----
+include::example$create-vector-search-index-header.sh[]
+----
+. Enter the JSON payload for the settings you want in your index. 
++
+Do not include the xref:search:search-index-params.adoc#uuid[uuid] or xref:search:search-index-params.adoc#sourceuuid[sourceUUID] parameters. 
++
+TIP: If you remove the xref:search:search-index-params.adoc#uuid[uuid] and xref:search:search-index-params.adoc#sourceuuid[sourceUUID] parameters, you can copy the Search index definition JSON payload from the Capella UI to use in a REST API call.
```
Comment (lamagnifica) [inline modules/vector-search/pages/create-vector-search-index-rest-api.adoc]: I missed any mention of the JSON payload in the Quick Mode topic. Is that the same as creating it with the UI?
I feel like I'm missing 
- all the advanced setting / config info
- how to test/the results of setting up via the UI, presumably with a payload that can be reused via cURL

File: modules/vector-search/pages/create-vector-search-index-ui.adoc [hunk 24]
```diff
@@ -0,0 +1,74 @@
...
+
+== Procedure 
+
+To create a Vector Search index with Quick Mode in Capella:
+
+. On the *Databases* page, select the database where you want to create a Search index.
+. Go to menu:Data Tools[Search].
+. Click btn:[Create Search Index].
++
+By default, *Quick Mode* should be on.
```
Comment (lamagnifica) [inline modules/vector-search/pages/create-vector-search-index-ui.adoc:42]: is this where you might switch to Advanced Mode?

**Review by dgnicholson — CHANGES_REQUESTED** [review]
This is a monster, but really well done @sarahlwelton. 🚀 

I focused mostly on structure, functionality, and readability and my only main concerns are around several broken links, but otherwise I think this is good to go. Anything with "Nit" is at your discretion.

File: modules/vector-search/pages/run-vector-search-ui.adoc [hunk 25]
```diff
@@ -0,0 +1,43 @@
+= Run A Vector Search with the Capella UI
```
Comment (lamagnifica) [inline modules/vector-search/pages/run-vector-search-ui.adoc]: lc A

File: modules/vector-search/pages/run-vector-search-ui.adoc [hunk 26]
```diff
@@ -0,0 +1,43 @@
+= Run A Vector Search with the Capella UI 
+:page-topic-type: guide
+:description: Run a Vector Search query from the Couchbase Capella UI to preview the search results from a Vector Search index.
```
Comment (lamagnifica) [inline modules/vector-search/pages/run-vector-search-ui.adoc]: is this a verification/testing step? that is, why would I want to preview?

File: modules/vector-search/pages/run-vector-search-ui.adoc [hunk 27]
```diff
@@ -0,0 +1,43 @@
...
+. Go to menu:Data Tools[Search]. 
+. Next to your Vector Search index, click btn:[Search].
+. In the *Search* field, enter a search query. 
++
+For example, the following query searches for the top 2 vectors similar to the vector `[ 0.707106781186548, 0, 0.707106781186548 ]` in the `colorvect_dot` field:
++
+[source,json]
+----
+include::example$run-vector-search-payload-ui.jsonc[]
+----
```
Comment (lamagnifica) [inline modules/vector-search/pages/run-vector-search-ui.adoc:42]: I haven't seen the UI, but this strikes me as very different from other queries in Capella. 
I would expect first a SQL++-looking query, and then the JSON result, rather than both together in one JSON -formatted document
Just raising that as possibly something to state explicitly, that these queries are not SQL++ queries, and whatever steps they may need to take to embed the query in JSON formatting (and when to expect the result to get added to the JSON)

File: modules/vector-search/pages/run-vector-search-ui.adoc [hunk 28]
```diff
@@ -0,0 +1,43 @@
...
++
+By using the special `match_none` query in the `query` field, the Search query only returns the `k` number of similar vectors.
+
+. Press kbd:[Enter].
```
Comment (lamagnifica) [inline modules/vector-search/pages/run-vector-search-ui.adoc]: [EDIT disregard, see next comment]
...and now I'm not sure what I was accomplishing by doing these steps. I assumed this was a test, a way to see if what you set up was enabled and giving responses to queries, but now it's almost like you're training it?
Is there no response to the Enter?

File: modules/vector-search/pages/run-vector-search-ui.adoc (hunk 22 above)
Comment (lamagnifica) [inline modules/vector-search/pages/run-vector-search-ui.adoc]: the links in preview are broken, so I opened the current Capella doc and found more info about Run A Simple Search with the Capella UI and Search Request JSON Properties, so never mind my confusion in this topic. Possibly just a reassuring sort of, Running a vector search index query in the Capella UI is similar to running a simple search (xref) for those of us who dropped in from another planet...

File: modules/vector-search/pages/create-vector-search-index-rest-api.adoc [hunk 29]
```diff
@@ -0,0 +1,59 @@
+= Create a Vector Search Index with the REST API and curl/HTTP
```
Comment (lamagnifica) [inline modules/vector-search/pages/create-vector-search-index-rest-api.adoc:1]: I've always used cURL -- not a cb standard?

**Review by lamagnifica — COMMENTED** [review]
Coming in as a completely naive user (no vector search background OR Capella search background), there were a few confusing points, though for the most part just a few nits and suggestions for rephrasing. Two thumbs up!

File: modules/search/partials/vector-search-field-descriptions.adoc (hunk 2 above)
Comment (sarahlwelton) [reply modules/search/partials/vector-search-field-descriptions.adoc]: Fair! Fixed.

File: modules/vector-search/pages/create-vector-search-index-ui.adoc (hunk 7 above)
Comment (sarahlwelton) [reply modules/vector-search/pages/create-vector-search-index-ui.adoc]: Thought I caught all these from my copy/paste - thanks!

File: modules/vector-search/pages/create-vector-search-index-rest-api.adoc (hunk 17 above)
Comment (sarahlwelton) [reply modules/vector-search/pages/create-vector-search-index-rest-api.adoc]: Good shout.

File: modules/vector-search/pages/create-vector-search-index-rest-api.adoc (hunk 29 above)
Comment (sarahlwelton) [reply modules/vector-search/pages/create-vector-search-index-rest-api.adoc:1]: We had a whole discussion about this when I first started doing this and apparently even the guy who created curl doesn't write it as cURL anymore? Who knew

File: modules/vector-search/pages/create-vector-search-index-rest-api.adoc (hunk 19 above)
Comment (sarahlwelton) [reply modules/vector-search/pages/create-vector-search-index-rest-api.adoc:35]: Surprisingly they do refer to it as a "verb"

File: modules/vector-search/pages/create-vector-search-index-rest-api.adoc (hunk 20 above)
Comment (sarahlwelton) [reply modules/vector-search/pages/create-vector-search-index-rest-api.adoc]: This depends on which endpoint you're using.

File: modules/vector-search/pages/create-vector-search-index-rest-api.adoc (hunk 23 above)
Comment (sarahlwelton) [reply modules/vector-search/pages/create-vector-search-index-rest-api.adoc]: Quick Mode does not expose you to the JSON payload at all, unless you want to view it. It's hidden. Quick Mode is a mode avilable in the UI. That's why the prereqs include "logging into the Couchbase Capella UI." 

Advanced settings are kept in another topic so users can just get the full breakdown of this huge JSON payload. 

Testing is just running a search and seeing if you get the results you want.

File: modules/vector-search/pages/create-vector-search-index-rest-api.adoc (hunk 21 above)
Comment (sarahlwelton) [reply modules/vector-search/pages/create-vector-search-index-rest-api.adoc]: I like this idea.

File: modules/vector-search/pages/create-vector-search-index-ui.adoc (hunk 24 above)
Comment (sarahlwelton) [reply modules/vector-search/pages/create-vector-search-index-ui.adoc:42]: Yes

File: modules/vector-search/pages/create-vector-search-index-ui.adoc (hunk 13 above)
Comment (sarahlwelton) [reply modules/vector-search/pages/create-vector-search-index-ui.adoc]: No workaround - what would you suggest as an alternative?

File: modules/vector-search/pages/create-vector-search-index-ui.adoc (hunk 14 above)
Comment (sarahlwelton) [reply modules/vector-search/pages/create-vector-search-index-ui.adoc:59]: I added a note to the previous step - let me know if that resolves your question?

File: modules/vector-search/pages/create-vector-search-index-ui.adoc (hunk 16 above)
Comment (sarahlwelton) [reply modules/vector-search/pages/create-vector-search-index-ui.adoc]: Yes, this is the child field you created a type mapping for.

File: modules/vector-search/pages/create-vector-search-index-ui.adoc (hunk 15 above)
Comment (sarahlwelton) [reply modules/vector-search/pages/create-vector-search-index-ui.adoc]: Yes

File: modules/vector-search/pages/run-vector-search-ui.adoc (hunk 26 above)
Comment (sarahlwelton) [reply modules/vector-search/pages/run-vector-search-ui.adoc]: "preview and test"?

File: modules/vector-search/pages/run-vector-search-ui.adoc (hunk 27 above)
Comment (sarahlwelton) [reply modules/vector-search/pages/run-vector-search-ui.adoc:42]: Because it is very different from all other queries in Capella. This isn't based on SQL++ at all - it's full text search. 

This also isn't the result, but the actual query you need to put into the Search field to return a result.

File: modules/vector-search/partials/nav.adoc (hunk 6 above)
Comment (sarahlwelton) [reply modules/vector-search/partials/nav.adoc:5]: I wasn't sure if the API instructions were going to get too into the weeds, and honestly... It's just the complexity of the whole thing. 

I wanted to follow the structure I set up in basic search by having things separate. There was concerns about how hard search is to parse and understand, that I wanted to keep things dead-ass simple for the UI and then offer some help to get around the awkward API documentation with an explicit example. 

Tabs could be something to consider, but I also think there's something to be said for simplicity - especially with the differing prerequisites between UI and REST.

File: modules/vector-search/pages/create-vector-search-index-ui.adoc (hunk 13 above)
Comment (lamagnifica) [reply modules/vector-search/pages/create-vector-search-index-ui.adoc]: Vectors must be represented as floating point numbers.

File: modules/vector-search/pages/create-vector-search-index-ui.adoc (hunk 14 above)
Comment (lamagnifica) [reply modules/vector-search/pages/create-vector-search-index-ui.adoc:59]: I guess I still don't understand if there's a new field (child or otherwise) being added or if I'm just adding configuration to something I selected.

Comment (sarahlwelton) [reply modules/vector-search/pages/create-vector-search-index-ui.adoc:59]: So, you have the document schema displayed. You select the field from the JSON schema that contains the data you want to be able to search from your index. 

That's the child field, the field you selected. 

Then, you're adding configuration onto that. 

You select the field on one side, and configure on the other.

File: modules/n1ql/pages/n1ql-language-reference/searchfun.adoc [hunk 30]
```diff
@@ -1,54 +1,54 @@
...
-Search functions enable you to use full text search (xref:server:fts:full-text-intro.adoc[FTS]) queries directly within a {sqlpp} query.
+Search functions enable you to use xref:search:search.adoc[Full Text Search (FTS)] queries directly within a {sqlpp} query.
 
 [float]
 === Prerequisites
 
 To use any of the search functions, the Search service must be available on the cluster.
-It is also recommended, but not required, that you should create suitable full text indexes for the searches that you need to perform.
-For more information, refer to xref:server:fts:fts-quickstart-guide.adoc#preparing-for-full-text-searches[Preparing for Full Text Searches].
+It is also recommended, but not required, that you should create suitable Search indexes for the searches that you need to perform.
```
Comment (ElliotFrancisHunter) [inline modules/n1ql/pages/n1ql-language-reference/searchfun.adoc]: This feels a little awkward on the scan, can we cut the 'but not required' out and still carry the intent across?

File: modules/search/pages/search.adoc [hunk 31]
```diff
@@ -52,9 +52,19 @@ You can run a Search query:
...
+
+For more information about Vector Search, see xref:vector-search:vector-search.adoc[].
+
 == See Also 
 
 * xref:create-search-indexes.adoc[]
 * xref:customize-index.adoc[]
 * xref:index-aliases.adoc[]
 * xref:run-searches.adoc[]
+* xref:vector-search:vector-search.adoc[]
```
Comment (ElliotFrancisHunter) [inline modules/search/pages/search.adoc:72]: Was it intentional to have two links to the same page close to each other? I understand the usefulness, just confirming this was deliberate :)

File: modules/search/pages/customize-index.adoc [hunk 32]
```diff
@@ -1,51 +1,69 @@
-= Customize a Search Index with the Capella UI
+= Search Index Features
```
Comment (ElliotFrancisHunter) [inline modules/search/pages/customize-index.adoc:1]: Seconding Daniel's thoughts, succinct and useful titles throughout. Awesome stuff!

File: modules/search/pages/quick-index-field-options.adoc [hunk 33]
```diff
@@ -0,0 +1,129 @@
+= Quick Index Mapping and Field Options 
+:page-topic-type: reference
+:description: When you create a Search index with Quick Mode in the Capella UI, you must set options for each field or type mapping you add to the index. 
+
+[abstract]
+{description}
+
+You can create the following types of type mappings in Quick Mode: 
+
+* <<quick-collection,>>
```
Comment (ElliotFrancisHunter) [inline modules/search/pages/quick-index-field-options.adoc:12]: Remove commas

**Review by ElliotFrancisHunter — COMMENTED** [review]
This is some really awesome work! Other than a handful of queries there wasn't much I could find issue-wise. Especially impressive as this PR is humungous 😱

File: modules/search/pages/create-quick-index.adoc [hunk 34]
```diff
@@ -0,0 +1,56 @@
...
+. Click btn:[Create Search Index].
++
+By default, *Quick Mode* should be on.
+. In the *Index Name* field, enter a name for the index. 
++
+[NOTE]
+====
+Your index name must start with an alphabetic character (a-z or A-Z). It can only contain alphanumeric characters (a-z, A-Z, or 0-9), hyphens (-), or underscores (_).
+
+For version 7.6 and later, your index name must be unique inside your selected bucket and scope. You cannot have 2 indexes with the same name inside the same bucket and scope.
```
Comment (bfavini) [inline modules/search/pages/create-quick-index.adoc]: Should probably say "For Couchbase Server 7.6 and later"

Comment (sarahlwelton) [reply modules/search/pages/create-quick-index.adoc]: Even though this is Capella? I'm torn on how to refer to the version.

File: modules/vector-search/pages/vector-search.adoc [hunk 35]
```diff
@@ -0,0 +1,50 @@
...
+
+[abstract]
+{description}
+
+== About Vector Search
+
+Vector Search builds on Couchbase Capella's xref:search:search.adoc[Search Service] to provide vector index support.
+You can use these new Vector Search indexes for Retrieval Augmented Generation (RAG) with an existing Large Language Model (LLM). 
+
+Using Vector Search, Capella, an embedding model, and your chosen LLM, you can develop AI applications while giving context and up-to-date information from your own data.
```
Comment (bfavini) [inline modules/vector-search/pages/vector-search.adoc]: Agree we don't want to get into the weeds...but it feels like we could use a bit more context around RAG, LLM, and embedding models. I would not hold up GA for this, but I think some simple definitions of these things and the role they play in AI would be helpful, particularly to new developers

File: modules/search/pages/create-quick-index.adoc [hunk 36]
```diff
@@ -0,0 +1,56 @@
+= Create a Search Index in Quick Mode
+:page-topic-type: guide
+:description: Use Quick Mode to create a Search index in Couchbase Capella with streamlined options.
+
+[abstract]
+{description}
+
+Use xref:create-search-index-ui.adoc[Advanced Mode] to add additional settings and more advanced options for your Search index and xref:customize-index.adoc#type-mappings[type mappings].
```
Comment (bfavini) [inline modules/search/pages/create-quick-index.adoc]: Would be even more people to help people understand what exactly they're giving up when they choose quick mode or when it's the right choice.....(Choose Quick Mode when....)

File: modules/vector-search/pages/create-vector-search-index-ui.adoc [hunk 37]
```diff
@@ -0,0 +1,78 @@
+= Create a Vector Search Index in Quick Mode
+:page-topic-type: guide
+:description: Use Quick Mode to create a Vector Search index in Couchbase Capella.
+
```
Comment (bfavini) [inline modules/vector-search/pages/create-vector-search-index-ui.adoc:6]: Similar comment in the general FTS section--would be helpful to call out what Quick Mode doesn't give me...why/when would I choose Quick Mode over Advanced and vice versa?

File: modules/vector-search/pages/create-vector-search-index-ui.adoc [hunk 38]
```diff
@@ -0,0 +1,78 @@
...
+. Click btn:[Create Search Index].
++
+By default, *Quick Mode* should be on.
+. In the *Index Name* field, enter a name for the Vector Search index. 
++
+[NOTE]
+====
+Your index name must start with an alphabetic character (a-z or A-Z). It can only contain alphanumeric characters (a-z, A-Z, or 0-9), hyphens (-), or underscores (_).
+
+For version 7.6 and later, your index name must be unique inside your selected bucket and scope. You cannot have 2 indexes with the same name inside the same bucket and scope.
```
Comment (bfavini) [inline modules/vector-search/pages/create-vector-search-index-ui.adoc]: Couchbase Server version7.6 or later

File: modules/vector-search/pages/create-vector-search-index-ui.adoc [hunk 39]
```diff
@@ -0,0 +1,78 @@
...
+====
+. Under *Type Mappings*, in the *Bucket* list, select the bucket that contains the documents you want to include in your index.
```
Comment (bfavini) [inline modules/vector-search/pages/create-vector-search-index-ui.adoc:51]: I originally intended to recommend using a different structure here, removing some of the conceptual or reference information from the steps. In the spirit of NOT documenting how to use the UI.

However, having read and reread this several times, I don't think you can avoid it. The procedure in the UI is just too complex/tricky.

**Review by bfavini — APPROVED** [review]
Nice job on this massive PR. Some general comments to enhance moving forward. Nothing should hold up GA.

File: modules/n1ql/pages/n1ql-language-reference/searchfun.adoc (hunk 30 above)
Comment (sarahlwelton) [reply modules/n1ql/pages/n1ql-language-reference/searchfun.adoc]: I didn't write this but I'm sure we could make that change.

File: modules/search/pages/quick-index-field-options.adoc (hunk 33 above)
Comment (sarahlwelton) [reply modules/search/pages/quick-index-field-options.adoc:12]: It's part of the markup, I'm going to leave that in.

File: modules/search/pages/search.adoc (hunk 31 above)
Comment (sarahlwelton) [reply modules/search/pages/search.adoc:72]: Yup, deliberate! Just in case people skip on by without reading.

**Review by dgnicholson — APPROVED** [review]
